scripts/run_integration_tests_windows.sh

```diff
@@ -42,9 +42,20 @@ strip target/$BUILD_TARGET/release/cloud-hypervisor
 
 export RUST_BACKTRACE=1
 
-# Only run with 1 thread to avoid tests interfering with one another because
-# Windows has a static IP configured
-time cargo test $features_test "tests::windows::$test_filter"
+CARGO_ARGS=
+# On a desktop class CPU invoking all the available tests at once is likely
+# to cause cryptic false positives due to the resource limits. Reduce the
+# load automatically to 1/3 of the processor capacity to retain balance between
```

Comment:
Why did you choose 1/3? I mean it depends on the tests themselves. Let's say you have a test that runs a VM with 10 vCPUs and 3 other tests running with 1 vCPU each, your patch wouldn't solve this problem, right?

Reply:
If 1 test uses 10 vCPUs, probably it can come hard, that's right. The value of 1/3 is what i see working based on the checks using the existing tests on a a machine with 8 and another one with 16 CPUs. There it seems solved this way, 1/2 already appears to much.

How it exists today, one probably would be able to overload the `--test-filter` to pass the number of threads. Perhaps one could introduce an explicit option for the number of threads, if that sounds cleaner?

Thanks

Reply:
I think the correct fix is to identify why the tests fail when under the constrained demand. We used to have such a `--test-threads` limit on the main integration tests but it was removed when the test stability was improved.

Reply:
From what I've seen the reason is as simple as the resource shortage. Fe right now the 7 tests running simultaneously on a  i7-10700K (8 core, 16 threads) and 32G RAM produce a load of ~42 on a Linux desktop machine. Even though kernels on desktop systems are usually preemptive, it is most likely the task scheduler being not happy with this load.

Perhaps what it is about is a performance issue that the parallel testing uncovers. I could sure dig further on this. For a baseline - what would be the simultaneous number of tests we'd expect to run without an issue on a desktop machine? Say on the configuration like above and say every guest uses 2 vCPUS, spawning 7 guests should be actually not a big issue. At least what to state is that most likely `dnsmasq` is of no issue here, it's rather about the parallel test vs. resources.

Thanks.